src/App.tsx

```diff
@@ -14,37 +14,85 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+enum SortType {
+  None = '',
+  Alphabetical = 'alphabetical',
+  Length = 'length',
+}
 
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState<SortType>(SortType.None);
```

Comment:
In this case the explicit generic parameter is not strictly necessary because TS can infer the type from the initial value. BUT it is good practice to include it for clarity and type safety👍🏻